unhandled promise rejection in simulator

```
[ERROR] encountered unhandled promise rejection at process level: [object Promise], reason: Error: GET http://127.0.0.1:53705/raw/helix/github.com--davidnuescheler--pages/master/helix-markup.yaml aborted
    at abortError (.hlx/pages/master/node_modules/fetch-h2/dist/lib/fetch-common.js:132:16)
    at AbortSignalImpl.abortHandler (.hlx/pages/master/node_modules/fetch-h2/dist/lib/fetch-common.js:141:28)
    at Object.onceWrapper (events.js:286:20)
    at AbortSignalImpl.emit (events.js:198:13)
    at AbortSignalImpl.EventEmitter.emit (domain.js:448:20)
    at AbortSignalImpl.once (.hlx/pages/master/node_modules/fetch-h2/dist/lib/abort.js:13:18)
    at Object.onceWrapper (events.js:286:20)
    at AbortSignalImpl.emit (events.js:198:13)
    at AbortSignalImpl.EventEmitter.emit (domain.js:448:20)
    at AbortController.abort (.hlx/pages/master/node_modules/fetch-h2/dist/lib/abort.js:22:25)
[hlx] error: Gateway timeout of 1000 milliseconds exceeded for http://127.0.0.1:53705/raw/helix/github.com--davidnuescheler--pages/master/helix-markup.yaml
```

--- a/src/utils/Downloader.js
+++ b/src/utils/Downloader.js
@@ -185,6 +185,9 @@ class Downloader {
     task.uri = uri;
     task.id = opts.id || '';
     task.options = opts;
+    // ignore rejections in case the task is never `awaited`. this avoids unhandled promise
+    // rejection warnings
+    task.catch(() => {});
 
     this._queue.push(task);
     return task;

--- a/test/testHTML.js
+++ b/test/testHTML.js
@@ -595,6 +595,67 @@ ${context.content.document.body.innerHTML}`,
     assert.ok(res.body.match(/<img/));
   });
 
+  it('html.pipe makes HTTP requests and handles timeouts', async function test() {
+    const { server } = this.polly;
+    server
+      .get('https://raw.githubusercontent.com/trieloff/soupdemo/master/helix-markup.yaml')
+      .intercept(async (req, res) => {
+        // the timeout for github is set to 100ms, so we cause a timeout here
+        await server.timeout(20000);
+        res.sendStatus(200);
+      });
+    server
+      .get('https://adobeioruntime.net/api/v1/web/helix/helix-services/content-proxy@v2?owner=trieloff&repo=soupdemo&path=%2Fhello.md&ref=master')
+      .intercept(async (req, res) => {
+        // the timeout for content proxy is 20s. so we make it a bit longer than above. this causes
+        // a promise rejection in the 'markupconfig' task.
+        await server.timeout(500);
+        res.sendStatus(200);
+      });
+
+    const result = await pipe(
+      (context) => {
+        const { content } = context;
+        // this is the main function (normally it would be the template function)
+        // but we use it to assert that pre-processing has happened
+        assert.ok(content.body);
+        assert.ok(content.mdast);
+        assert.ok(content.meta);
+        assert.ok(content.document);
+        assert.equal(typeof content.document.getElementsByTagName, 'function');
+        assert.equal(content.document.getElementsByTagName('h1').length, 1);
+        assert.equal(content.document.getElementsByTagName('h1')[0].innerHTML, 'Bill, Welcome to the future');
+        assert.equal(content.meta.template, 'Medium');
+        assert.equal(content.intro, 'Project Helix');
+        assert.equal(content.title, 'Bill, Welcome to the future');
+        assert.deepEqual(content.sources, ['https://raw.githubusercontent.com/trieloff/soupdemo/master/hello.md']);
+        // and return a different status code
+        context.response = { status: 201, body: content.document.body.innerHTML };
+      },
+      {
+        request: {
+          params: {
+          },
+        },
+      },
+      {
+        request: {
+          headers: { 'Cache-Control': 'no-store' },
+          params,
+        },
+        secrets: {
+          ...secrets,
+          HTTP_TIMEOUT: 100,
+        },
+        logger,
+      },
+    );
+    console.log(logger.getOutput());
+
+    const res = result.response;
+    assert.equal(res.status, 504);
+  });
+
   it('html.pipe makes HTTP requests and falls back to master', async () => {
     const myparams = { ...params };
     delete myparams.ref;

